juliasamartseva/ContactsArgumentParser/Sources/ContactsArgumentParser/main.swift

```
@@ -0,0 +1,298 @@
+import ArgumentParser
+import Contacts
+import Foundation
+
+struct Contacts: ParsableCommand {
+
+  private static var contacts: ContactList? = try? ContactList()
+
+  static var configuration = CommandConfiguration(
+    abstract: "A utility for working with contacts",
+    subcommands: [
+      AddContact.self, AddGroup.self, AddGroupToContact.self, ShowAllContacts.self,
+      ShowAllGroups.self, RemoveContact.self, RemoveGroup.self, ShowContact.self, ShowGroup.self,
+      ShowContactsInGroups.self, RemoveGroupFromContact.self,
+    ]
+  )
+}
+
+struct Options: ParsableArguments {
+  @Option
+  var name: String
+
+  @Option
+  var phone: String
+
+  @Option
+  var email: String?
+
+  @Option
+  var jobTitle: String?
+
+  @Option
+  var company: String?
+}
+
+extension Contacts {
+
+  struct AddContact: ParsableCommand {
+    static var configuration = CommandConfiguration(
+      abstract: "Add contact to existing contact list.")
+
+    @OptionGroup
+    var contactOption: Options
+
+    mutating func run() throws {
+      guard let contact = try initialiseContact(contactOption: contactOption) else {
+        print("Cannot initialise contact.")
+        return
+      }
+      try contacts?.add(contact: contact)
```

you may inline creation of the `contacts` here. Otherwise the error would be swallowed.